app/components/primer/alpha/banner.rb

```diff
@@ -121,6 +125,14 @@ def initialize(full: false, full_when_narrow: false, dismissible: false, descrip
 
       private
 
+      def before_render
+        @system_arguments[:classes] = class_names(
+          @system_arguments[:classes],
+          # Center with max-width for a lot of content or long descriptions
+          "Banner--centered": @full && content && content.length > 120 || @full && @description && @description.length > 120
```

Comment:
One concern with checking length vs explicitly setting it, is if someone adds breakpoints on their own to the banner. We all anything to be put into the content of the banner, what if it's something like this:

```erb
      <%= render(Primer::Alpha::Banner.new) do |component| %>
        <% component.with_action_button(size: :small) { "Take action" } %>
        This is a banner with an action button!<br/>
        We sometimes have banners with a lot of text.<br/>
        But in this case I'm breaking the text to make sure the line lengths are small.<br/>
        This can be easily done with the `<br>` tag.
      <% end %>
```

Reply:
Hmm.. Manually adding line breaks with `<br>` would behave like:

### Before 
![Screen Shot 2023-03-13 at 12 24 31](https://user-images.githubusercontent.com/378023/224601162-42ff078b-60c9-4f35-ad05-83402152fdfa.png)

### After
![Screen Shot 2023-03-13 at 12 24 42](https://user-images.githubusercontent.com/378023/224601165-5f168837-1ed0-4bba-9fda-cea81ccc2c60.png)

So yeah, it would make the content centered unrelated to the line length. Not sure if that's ok.

I think in general, it would be nice to avoid adding line breaks manually. Otherwise on narrow screens it could look weird once the the short lines start wrapping and create these gaps in between:

![Screen Shot 2023-03-13 at 12 16 38](https://user-images.githubusercontent.com/378023/224600990-a8e4b376-af29-46a8-87bc-48cd6a3f9e84.png)

I thought about adding a `centered` option that you can set manually. But that comes with the risk of people using it also for short lines.